File: 4107-find-missing-elements/find-missing-elements.java
import java.util.*;

class Solution{
    public List<Integer> findMissingElements(int[] nums){
        int n=nums.length;

        HashMap<Integer, Integer> mp=new HashMap<>();
        for(int num: nums){
            mp.put(num, mp.getOrDefault(num, 0)+1);
        }

        IntSummaryStatistics stats = Arrays.stream(nums).summaryStatistics();
        int mn=stats.getMin();
        int mx=stats.getMax();

        List<Integer> res=new ArrayList<>();
        int i=mn;
        while(i<=mx){
            if(!mp.containsKey(i)){
                res.add(i);
            }
            i++;
        }

        return res;

    }
}

File: Easy/4107-find-missing-elements/find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        n=len(nums)

        mp={}
        for num in nums:
            mp[num]=mp.get(num, 0)+1

        mx=max(nums)
        mn=min(nums)

        res=[]
        i=mn
        while i<=mx:
            if i not in mp:
                res.append(i)
            i+=1

        return res
